R/tspa.R

```diff
@@ -58,222 +111,205 @@
-#   }
-#   col <- colnames(data)
-#   fs <- paste0("fs_", var)
-#
-#   if (length(fs) < 2) {
-#     stop("Data column len is smaller than 2, unable to build model. Data needs to consist of at least 2 columns.");
-#   }
-#
-#   tspaModel <- rep(NA, len)
-#   latent_var <- rep(NA, len)
-#   error_constraint <- rep(NA, len)
-#   latent_variance <- rep(NA, len)
-#   reliability_constraint <- rep(NA, len)
-#
-#   if(fs %in% col == FALSE) {
-#     stop("data columns does not match with reliability")
-#   }
-#
-#   for (x in 1:len) {
-#     latent_var[x] <- paste0(var[x], " =~ 1 * ", fs[x], "\n")
-#     error_constraint[x] <- paste0(fs[x], " ~~ ev", x, " * ", fs[x], "\n")
-#     latent_variance[x] <- paste0(var[x], " ~~ v", x, " * ", var[x], "\n")
-#     reliability_constraint[x] <- paste0("v", x, " == ", toString(reliability[x]), " / ", toString(1 - reliability[x]), " * ev", x, "\n")
-#   }
-#
-#   latent_var_str <- paste(latent_var, collapse="")
-#   error_constraint_str <- paste(error_constraint, collapse="")
-#   latent_variance_str <- paste(latent_variance, collapse="")
-#   reliability_constraint_str <- paste(reliability_constraint, collapse="")
-#   tspaModel <- paste0(latent_var_str, error_constraint_str, latent_variance_str, model, "\n", reliability_constraint_str)
-#
-#   tspa_fit <- sem(model = tspaModel,
-#                   data  = data)
-#
-#   # tspaModel <- cat(tspaModel)
-#   attributes(tspa_fit)$tspaModel <- tspaModel
-#   # to access the attribute, use attr(x,"tspaModel")
-#   return (tspa_fit)
-# }
+  tspa_fit <- sem(model = tspaModel,
+                  data  = data,
+                  ...)
+  # to access the attribute, use attr(x,"tspaModel")
+  attr(tspa_fit, "tspaModel") <- tspaModel
+  return(tspa_fit)
+}
 
+tspaSingleGroup <- function(model, data, se = NULL) {
+  if (nrow(se) != 0) {
+    ev <- se^2
+    var <- names(se)
+    len <- length(se)
 
+    # col <- colnames(data)
+    fs <- paste0("fs_", var)
+    latent_var <- rep(NA, len)
+    error_constraint <- rep(NA, len)
+    # latent_variance <- rep(NA, len)
 
-tspa <- function(model, data, reliability = NULL, se = NULL, ...) {
-    if (is.null(reliability) == FALSE){
-        stop("tspa() currently does not support reliability model")
-    }
-    if (!is.data.frame(se)) {
-        se <- as.data.frame(as.list(se))
+    for (x in seq_len(len)) {
+      latent_var[x] <- paste0(var[x], " =~ ", fs[x], "\n")
+      error_constraint[x] <- paste0(fs[x], " ~~ ", ev[x], " * ", fs[x], "\n")
+      # latent_variance[x] <- paste0(var[x], " ~~ v", x, " * ", var[x], "\n")
     }
 
-    if(nrow(se) == 1){
-        tspaModel <- tspaSingleGroup(model, data, se)
-    }
-    else if(nrow(se) > 1){
-        tspaModel <- tspaMultipleGroupSe(model, data, se)
-    }
+    latent_var_str <- paste(latent_var, collapse = "")
+    error_constraint_str <- paste(error_constraint, collapse = "")
+    # latent_variance_str <- paste(latent_variance, collapse="")
```

Comment:
Do we need to remove the commented-out line?

Reply:
@winniewytse could you help checking whether the commented-out lines can be removed in this file? Thanks.